ggml/src/ggml-vulkan/ggml-vulkan.cpp

```diff
@@ -4621,6 +4621,29 @@ static void ggml_vk_instance_init() {
             }
         }
 
+        // Remove any unsupported devices
+        size_t num_unsupported = 0;
+        for (size_t i = 0; i < vk_instance.device_indices.size();) {
+            vk::PhysicalDevice vkdev = devices[vk_instance.device_indices[i]];
+
+            VkPhysicalDeviceFeatures2 device_features2;
+            device_features2.sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_FEATURES_2;
+
+            VkPhysicalDeviceVulkan11Features vk11_features;
+            vk11_features.pNext = nullptr;
+            vk11_features.sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_VULKAN_1_1_FEATURES;
+            device_features2.pNext = &vk11_features;
+
+            vkGetPhysicalDeviceFeatures2(vkdev, &device_features2);
+            if (!vk11_features.storageBuffer16BitAccess) {
```

Comment:
I think this check should move into a function (`bool ggml_vk_device_is_supported`), then you can add this check to the
```
if (new_props.properties.deviceType == vk::PhysicalDeviceType::eDiscreteGpu || new_props.properties.deviceType == vk::PhysicalDeviceType::eIntegratedGpu) {
```
in line 4543 above, instead of removing bad devices afterwards.